coreblocks/lsu/dummyLsu.py

```diff
@@ -217,6 +219,16 @@ def elaborate(self, platform):
         instr_is_load = Signal()
         m.d.comb += instr_is_load.eq(current_instr.exec_fn.op_type == OpType.LOAD)
 
+        addr = Signal(self.gen_params.isa.xlen)
+        m.d.comb += addr.eq(current_instr.s1_val + current_instr.imm)
+
+        pmas = Record(self.gen_params.get(PMALayouts).pma_attrs_layout)
+        m.d.comb += pma_checker.addr.eq(addr)
+        m.d.comb += pmas.eq(pma_checker.result)
+
+        is_mmio = Signal()
+        m.d.comb += is_mmio.eq(pmas["mmio"])
```

Comment:
Combinational `Signal` has two main roles:

* Being used together with Amaranth's control structures (`m.If` etc.)
* De-duplicating expressions (as Amaranth works on trees instead of DAGs, unfortunately).

This signal is used for none of the above purposes, so a regular Python variable is enough:

```suggestion
        is_mmio = pmas["mmio"]
```

But actually, the RHS is so short and descriptive I would just consider inlining the definition at use site.